### PR TITLE
[YPPY-1615] - Get release process working

### PR DESCRIPTION
**Pull request by FalconerTC**
Setup github action to automatically release and deploy.

Runs after successful merge to master rather than push to master, to avoid loops with maven-release-plugin

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -18,21 +20,41 @@ jobs:
...
+            "password": "${{ secrets.OSS_NEXUS_PASSWORD }}"
+          }]'
+
+      - name: Setup Github SSH key
+        uses: webfactory/ssh-agent@v0.4.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_KEY }}
 
-          maven-args: "--no-transfer-progress"
+      - name: Prepare release
```
Comment (gtrog) [inline .github/workflows/release.yml:56]: Very nice!  What happens if there is a failure, can we add a `mvn release:rollback` step for good measure if there's failure, so we don't leave it in a half-released state?

File: README.md [hunk 2]
```diff
@@ -48,3 +48,8 @@ If we can hit that bullseye, the rest of the dominoes will fall like a house of
 
 As of July 2019, Java 8 is supported.
 Support for Java 6 and Java 7 has ended.
+
+## Release process
+When a PR is merged to master, a release will be made and deployed to Sonatype by Github Actions. This includes pushing new commits via `maven-release-plugin`. The released version will be automatically deployed and usable immediately after the `release` workflow completes.
+
+To use the manual release process, see https://renttherunway.jira.com/wiki/spaces/EN/pages/100991186/How+to+Release+Conduit
```
Comment (fscala) [inline README.md:55]: are you sure we want to have this wiki link in this repo?

File: README.md [hunk 3]
```diff
@@ -48,3 +48,8 @@ If we can hit that bullseye, the rest of the dominoes will fall like a house of
 
 As of July 2019, Java 8 is supported.
 Support for Java 6 and Java 7 has ended.
+
+## Release process
```
Comment (fscala) [inline README.md:52]: not your code, but above in the readme, the method signatures have very odd formatting with the prefix comma

File: README.md (hunk 2 above)
Comment (FalconerTC) [reply README.md:55]: I'll update the page to indicate that it's not the preferred process anymore but it is still the fall-back so probably useful to include.

**Pull request merged**